proxy/reverse.go

```diff
@@ -98,7 +113,10 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		return
 	}
 
-	defer res.Body.Close()
+	defer func() {
+		completeCh <- true
```

Comment:
could we move this defer early? if it returns when res == nil, the go routine will not be recycled.